src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -335,8 +343,26 @@ private void validatePartitionProperties(FailureCollector collector) {
     }
   }
 
-  private void setInputFormat(BatchSourceContext context) {
+  private void setInputFormat(BatchSourceContext context,
+                              Schema configuredSchema) {
+    // Set input for Spark
     context.setInput(Input.of(config.referenceName, new BigQueryInputFormatProvider(configuration)));
+
+    // Add output for SQL Engine Direct read
+    ImmutableMap.Builder<String, String> arguments = new ImmutableMap.Builder<>();
+
+    List<String> fieldNames = configuredSchema.getFields().stream().map(f -> f.getName()).collect(Collectors.toList());
```

Comment:
We had to make a change on the BQ sink to only configure the Pushdown sink when a schema is configured in https://github.com/data-integrations/google-cloud/pull/1120/

Can you add the same here? Essentially, if `configuredSchema` is null, don't initialize the pushdown sink. Realistically, I don't see how pushdown execution would even work if the schema is not known at runtime.